src/metabase/analytics/snowplow.clj

```diff
@@ -175,7 +176,8 @@
        ;; Make sure keywords in payload are converted to strings in snake-case
        (m/map-kv
         (fn [k v] [(normalize-kw k) (if (keyword? v) (normalize-kw v) v)])
-        (assoc data :event event-kw))))
+        (merge {:event event-kw}
```

Comment:
we were overriding the `:event` in `data`. This is okay for a payload that doesn't have `:event` specified but actions have the `:event` in the payload, so merge last to make sure the specified event is preserved.